### ISSUE TITLE
Curiosity: Why did you make this?

### ISSUE THREAD

**Issue by spikespaz** [FractalBoy/perl-language-server#116]
It looks like [Rich Terger's language server](https://github.com/richterger/Perl-LanguageServer) is older. I would make a comparison, but yours is the only one in `nixpkgs`.

What are the differences, and why did you make this one rather than fixing Rich's?

Maybe this ought to be answered in the readme, why we need a new LSP server, for those who come looking for Perl tooling.

It works well, I applaud you.

**Comment by FractalBoy** [comment]
I think it was mostly because I wanted to learn/have a side project, and because of that I'd rather build something from scratch than basically rewrite another project.

**Comment by miversen33** [comment]
> It looks like [Rich Terger's language server](https://github.com/richterger/Perl-LanguageServer) is older. I would make a comparison, but yours is the only one in `nixpkgs`.
> 
> What are the differences, and why did you make this one rather than fixing Rich's?
> 
> Maybe this ought to be answered in the readme, why we need a new LSP server, for those who come looking for Perl tooling.
> 
> It works well, I applaud you.

I've used both and can say that setup of this language server is much easier and smoother than setup of Rich's. In addition, I feel like this one works better. Rich's is unmaintained, and IMO it's a really big ask to ask someone to maintain your codebase. It's almost always easier just to make your own (especially when the projects are small, such as these). I'm unsure there needs to be a reason why something is created, I've made plenty of things "just cause", and that's good enough reason to me. To add on to that, the fact that this one works well and is actively maintained and supported should be a good enough reason to use it over unmaintained codebases.

@spikespaz
